File: QuanLySanCauLong/lib/models/san_gio_response_dto.dart
class SanGioResponseDto {
  final int sanId;
  final String tenSan;
  final List<GioStatusDto> gios;

  SanGioResponseDto({required this.sanId, required this.tenSan, required this.gios});

  factory SanGioResponseDto.fromJson(Map<String, dynamic> json) {
    return SanGioResponseDto(
      sanId: json['SanId'],
      tenSan: json['TenSan'],
      gios: (json['Gios'] as List).map((e) => GioStatusDto.fromJson(e)).toList(),
    );
  }
}

class GioStatusDto {
  final int gioId;
  final String gio;
  final double donGia;
  final bool isBooked;

  GioStatusDto({required this.gioId, required this.gio, required this.donGia, required this.isBooked});

  factory GioStatusDto.fromJson(Map<String, dynamic> json) {
    return GioStatusDto(
      gioId: json['GioId'],
      gio: json['Gio'],
      donGia: json['DonGia'],
      isBooked: json['IsBooked'],
    );
  }
}

File: quan_ly_san_cau_long_API/quan_ly_san_cau_long_API/DTOs/SanGioResponseDto.cs
namespace quan_ly_san_cau_long_API.DTOs
{
    public class SanGioResponseDto
    {
        public int SanId { get; set; } // ID của sân
        public string TenSan { get; set; } // Tên sân
        public List<GioStatusDto> Gios { get; set; } // Danh sách giờ và trạng thái
    }

    public class GioStatusDto
    {
        public int GioId { get; set; } // ID giờ
        public string Gio { get; set; } // Thời gian (VD: "18:00")
        public double DonGia { get; set; } // Giá giờ chơi
        public bool IsBooked { get; set; } // Trạng thái: true nếu đã đặt, false nếu còn trống
    }

}
